Automattic/wp-calypso
# Modularise posts utils

`state/posts/utils` is currently a massive monolith of over 700 lines with a large amount of imports and exports. This makes tree-shaking difficult, causing sections that only need a small function to pull in all the dependencies.

This PR splits it into small modules of 1-2 exports, which are re-exported from an index.

Code changes are trivial, around lint errors and avoiding unnecessary use of arrow functions and function constants.

This PR won't produce any improvements by itself, as it needs the side-effect declaration added in #39625 for tree-shaking to work. If that PR gets submitted first, I will rebase this PR on top of it to get the numbers from ICFY.

#### Changes proposed in this Pull Request

* Split `state/posts/utils` into modules

#### Testing instructions

There are no functional code changes, so it should be sufficient to ensure that there are no build or unit test errors.
